include/biscuit/cpuinfo.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c), 2022, KNS Group LLC (YADRO)
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#pragma once
+
+#include <biscuit/assembler.hpp>
+#include <biscuit/registers.hpp>
+#include <cstddef>
+#include <cstdint>
+
+#include <sys/auxv.h>
+#include <sys/prctl.h>
+#include <asm/hwcap.h>
+
+namespace biscuit {
+
+#ifndef COMPAT_HWCAP_ISA_V
+#define COMPAT_HWCAP_ISA_V  (1 << ('V' - 'A'))
+#endif
+
+enum riscv_extension_t : uint64_t {
```

Comment:
With our naming convention this should be `RISCVExtension`.

This should also be an enum class (with bitwise operators overloaded if they need to be)

Reply:
Fixed the name, replaced with enum class